packages/react-components/src/components/Picker/Picker.tsx

```diff
@@ -70,32 +78,96 @@ export const Picker: React.FC<IPickerProps> = ({
     }
   }, [isListOpen]);
 
-  const handleOnTriggerClick = () => {
-    if (disabled) {
+  React.useEffect(() => onSelect(selectedItems), [selectedItems]);
```

Comment:
can't we call `onSelect` in event handlers? And do we really need the internal state of the selected items?

Reply:
After some analysis, no. I've removed the internal state for `selectedItems` and from now, the component is fully controlled by external props passed inside.